src/soep_preparation/dataset_merging/task.py

```diff
@@ -33,216 +58,147 @@ def _dataset_level(dataset: pd.DataFrame) -> str:
     raise ValueError(msg)
 
 
-def _reorder_dict(original_dict, priority_keys):
-    # Create a dictionary where *priority_keys* come first
-    ordered_dict = {
-        key: original_dict[key] for key in priority_keys if key in original_dict
-    }
-
-    # Add the remaining keys that were not in the priority list
-    ordered_dict.update(
-        {key: original_dict[key] for key in original_dict if key not in ordered_dict},
+def _get_shared_columns(left_df: pd.DataFrame, right_df: pd.DataFrame) -> list:
```

Comment:
```suggestion
def _get_shared_merge_keys(left_df: pd.DataFrame, right_df: pd.DataFrame) -> list:
```
But see below on meta-information.